src/ReservoirPair.sol

```diff
@@ -331,23 +336,22 @@ abstract contract ReservoirPair is ReservoirERC20 {
         }
     }
 
-    function _syncManaged(uint104 aReserve0, uint104 aReserve1)
+    function _syncManaged(uint256 aReserve0, uint256 aReserve1)
         internal
-        returns (uint104 rReserve0, uint104 rReserve1)
+        returns (uint256 rReserve0, uint256 rReserve1)
     {
         if (address(assetManager) == address(0)) {
-            // PERF: Is assigning to rReserve0 cheaper?
             return (aReserve0, aReserve1);
         }
 
-        uint104 lToken0Managed = assetManager.getBalance(this, token0);
-        uint104 lToken1Managed = assetManager.getBalance(this, token1);
+        uint256 lToken0Managed = assetManager.getBalance(this, token0);
+        uint256 lToken1Managed = assetManager.getBalance(this, token1);
 
         rReserve0 = _handleReport(token0, aReserve0, token0Managed, lToken0Managed);
         rReserve1 = _handleReport(token1, aReserve1, token1Managed, lToken1Managed);
 
-        token0Managed = lToken0Managed;
-        token1Managed = lToken1Managed;
+        token0Managed = lToken0Managed.toUint104();
+        token1Managed = lToken1Managed.toUint104();
```

Comment:
I think it's appropriate to do a safecast over here since there's no guarantee that the new balance will be <= uint104